### PR TITLE
DFM - FSC

### PR DESCRIPTION
**Pull request by msohailhussain**
### Summary
- Added datafile manager FSC support
- Fixed an issue where notification was being sent for hardcoded config-updates, only it should be done for remote

### Issues
- Hardcoded datafile issue is not fixed, will fix later since can't debug on travis and locally it's working fine. 

### Test plan
- FSC are running with DFM.

### CODE REVIEW

File: scripts/setup-integration-tests.sh [hunk 1]
```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
+
+. ~/.nvm/nvm.sh
```
Comment (msohailhussain) [inline scripts/setup-integration-tests.sh]: We don't really need it.

File: scripts/setup-integration-tests.sh [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+mkdir -p $FSC_PATH
+pushd $FSC_PATH && git init && git fetch --depth=1 https://$CI_USER_TOKEN@github.com/optimizely/fullstack-sdk-compatibility-suite ${FSC_BRANCH:-master} && git checkout FETCH_HEAD
```
Comment (msohailhussain) [inline scripts/setup-integration-tests.sh]: I think it should not have inner `pushd`

File: tests/integration/optlyplugins/notification_manager.go [hunk 3]
```diff
@@ -25,9 +27,13 @@ import (
 	"github.com/optimizely/go-sdk/tests/integration/models"
 )
 
+// DefaultInitializationTimeout defines default timeout for datafile sync
+const DefaultInitializationTimeout = time.Duration(5000) * time.Millisecond
```
[the remark this answers is not in the document]
Comment (mikeng13) [reply tests/integration/optlyplugins/notification_manager.go]: 5 * time.Second

File: tests/integration/optlyplugins/utils.go [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+	"github.com/optimizely/go-sdk/pkg/registry"
+	"github.com/optimizely/go-sdk/tests/integration/models"
+)
+
+const localDatafileURLTemplate = "http://localhost:3001/datafiles/%s.json?request_id="
+
+// SyncConfig doesn't request for new datafile if we provide a valid datafile
+// this requires us to keep defaultPollingInterval low so that the request
+// initiated from Start method is executed quickly
+const defaultPollingInterval = time.Duration(1000) * time.Millisecond
```
[the remark this answers is not in the document]
Comment (pawels-optimizely) [reply tests/integration/optlyplugins/utils.go]: or just time.Second

**Pull request closed without merging**